### ISSUE TITLE
Inner progress bars displayed after outer progress finishes

### ISSUE THREAD

**Issue by andrewGhazi** [HenrikBengtsson/progressr#136]
I'm working on a package that has two potentially long running functions, both of which I would like to set up to take advantage of progressr. The second function repeatedly calls the first. In the following mock example, calling `f2()` correctly shows progress for the outer `map()`, but after completing it then displays a single "inner" progress bar. I tested this on both Windows and Linux.

I assume it's capturing all of the inner progress bars then showing them all simultaneously at the end once the outer map is complete? The behavior I was expecting was the outer progress bar to show, but not the inner bars. 

I know that nested progress notifications aren't supported, but I just want the inner progress bars to turn off when called from `f2()` (yet still be available when calling `f1()` directly). Is it possible from the developer side to tell a progressor to not show if there are higher level progressors operating? Or is the solution just to give another argument to `f1()` that can turn off the progressor?

Apologies if I missed any existing issues on this question, I couldn't find any.

```r
library(purrr)
library(progressr)
handlers(global = TRUE)

f1 = function(x = 5) {
  p = progressr::progressor(steps = x)
  
  map(1:x, ~{Sys.sleep(.2)
             p()
             .x})
}

f2 = function(x = 10) {
  p = progressr::progressor(steps = x)
  map(1:x, ~{p()
             f1()})
}

f1()
f2()
```

**Comment by HenrikBengtsson** [comment]
Thanks for reporting.  Here's a minimal base-R example that illustrates the problem:

```r
library(progressr)
handlers(global = TRUE)

handlers(handler_progress(clear = FALSE))

f_inner <- function(n = 4) {
  p <- progressr::progressor(n)
  for (kk in seq_len(n)) {
    Sys.sleep(0.1)
    p(sprintf("inner #%d", kk))
  }
}

f_outer <- function(n = 6) {
  p <- progressr::progressor(n)
  for (kk in seq_len(n)) {
    p(sprintf("outer #%d", kk))
    f_inner()
  }
}

f_outer()
```

which will output:

```r
| [=========================================] 100% outer #6
- [=========================================] 100% inner #4
```

**Comment by HenrikBengtsson** [comment]
TL;DR: Use

```r
f2 = function(x = 10) {
  p = progressr::progressor(steps = x)
  map(1:x, ~{f1()
             p()})
}
```

to avoid this. 

This is because with:

```r
f_outer <- function(n = 6) {
  p <- progressr::progressor(n)
  for (kk in seq_len(n)) {
    p(sprintf("outer #%d", kk))
    f_inner()
  }
}
```

the progressor `p` _completes_ when calling `p(sprintf("outer #%d", kk))` at the last `kk = n` iteration.  When it completes, it also "auto finishes", i.e. it removes itself automatically, letting any following progressor to take over.  In this example, calling `f_inner()` immediately after, will create a new progressor and there is no longer an "outer" progressor blocking it's signals from propagating all the way to the progress handler.

**Comment by andrewGhazi** [comment]
Ah okay, that makes sense, thank you! Putting `p()` after the inner function call has the added benefit of updating the outer progressor _after_ each slow inner step, which probably aligns better with the expected behavior.
